File: ecolink-front/src/app/core/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { User } from '../models/index';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class UserService {
  private base = `${environment.apiUrl}/users`;
  constructor(private http: HttpClient) {}

  getById(id: number): Observable<User> {
    return this.http.get<User>(`${this.base}/${id}`);
  }
  getByOpenId(openId: string): Observable<User> {
    return this.http.get<User>(`${this.base}/openid/${openId}`);
  }
  createOrUpdate(user: Partial<User>): Observable<User> {
    return this.http.post<User>(this.base, user);
  }
  updatePoints(id: number, points: number): Observable<User> {
    const params = new HttpParams().set('points', points);
    return this.http.put<User>(`${this.base}/${id}/points`, null, { params });
  }
}

File: flutter_application_1/lib/services/user_service.dart
import '../models/user.dart';
import 'api_client.dart';

class UserService {
  static Future<User> createOrUpdate(User user) async {
    final json = await ApiClient.post('/users', body: user.toJson());
    return User.fromJson(json as Map<String, dynamic>);
  }

  static Future<User?> getById(int id) async {
    final json = await ApiClient.get('/users/$id');
    return json != null ? User.fromJson(json as Map<String, dynamic>) : null;
  }

  static Future<User> updatePoints(int id, int points) async {
    final json = await ApiClient.put('/users/$id/points', query: {'points': points});
    return User.fromJson(json as Map<String, dynamic>);
  }
}
